spec/lib/classification_lifecycle_spec.rb

```diff
@@ -88,37 +76,66 @@
   describe "#transact!" do
     let(:classification) { create(:classification) }
 
-    describe "the classification lifecycle methods" do
+    after(:each) do
+      subject.transact! { true }
+    end
 
+    context "when the classification subjects have not been seen" do
       it "should wrap the calls in a transaction" do
-        subject
         expect(Classification).to receive(:transaction)
-        subject.transact!
       end
 
       it "should call the #mark_expert_classifier method" do
         expect(subject).to receive(:mark_expert_classifier).once
-        subject.transact!
       end
 
       it "should call the #update_seen_subjects method" do
         expect(subject).to receive(:update_seen_subjects).once
-        subject.transact!
       end
 
       it "should call the #dequeue_subject method" do
         expect(subject).to receive(:dequeue_subject).once
-        subject.transact!
       end
 
-      it "should call the instance_eval on the block if passed a block" do
+      it "should call the instance_eval on the passed block" do
         expect(subject).to receive(:instance_eval).once
-        subject.transact! { true }
       end
 
       it "should call the #publish_to_kafka method" do
         expect(subject).to receive(:publish_to_kafka).once
-        subject.transact!
+      end
+    end
+
+    context "when the classification subjects have not been seen" do
```

Comment:
can you change the wording to have been seen before or something like it? Please excuse the the pedantry but I had trouble reading this.